AutoROM/AutoROM.py

```diff
@@ -1,175 +1,188 @@
-        tar_file.write(chunk)
-    tar_file.close()
-
-# given the location of a ROMs.tar file, extract its contents into a singular folder
-def extract_tar_content(installation_dirs):
-    # extract tar files
-    # unzip each zip
-    # calculate checksum of each tar file
-    install_dir = installation_dirs[0]
-    with tarfile.open(install_dir+"Roms.tar.gz") as tar:
-        tar.extractall(install_dir)
-
-
-def transfer_rom_files(installation_dirs, checksum_map):
-    # go through every ROM in install_dir/delete/
-    # if the ROM file matches a checksum, store in install dir
-    install_dir = installation_dirs[0]
-    zip_dir = install_dir + "ROM/"
-    for subdir, _, files in os.walk(zip_dir):
-        for file in files:
-            hash_md5 = hashlib.md5()
-            with open(os.path.join(subdir, file), "rb") as f:
-                for chunk in iter(lambda: f.read(4096), b""):
-                    hash_md5.update(chunk)
-                d = str(hash_md5.hexdigest())
-                if d in checksum_map:
-                    # transfer file here to name in checksum map
-                    game_name = checksum_map[d][0:-4]
-                    game_subdir = install_dir+game_name+"/"
-                    if not os.path.exists(game_subdir):
-                        os.mkdir(game_subdir)
-                    os.rename(os.path.join(subdir, file), os.path.join(game_subdir, checksum_map[d]))
-                    del checksum_map[d]
-
-def clean_tar_files(installation_dirs):
-    # delete Roms.tar
-    # delete extracted HC ROMS.zip
-    # delete extracted ROMS.zip
-    # delete unzipped delete folder
-    install_dir = installation_dirs[0]
-    if os.path.exists(os.path.join(install_dir, "Roms.tar.gz")):
-        os.remove(os.path.join(install_dir, "Roms.tar.gz"))
-    if os.path.exists(os.path.join(install_dir, "ROM/")):
-        shutil.rmtree(os.path.join(install_dir, "ROM/"))
-
-def main(license_accepted=False, specific_dir=None):
-    ale_installed = True
-    multi_ale_installed = True
-    try:
-        import ale_py
-    except ImportError:
-        ale_installed = False
+def download_tar_to_buffer(url="https://roms8.s3.us-east-2.amazonaws.com/Roms.tar.gz"):
+    with requests.get(url, stream=True) as response:
+        assert response.status_code == 200
+        assert response.headers["Content-Type"] == "application/x-gzip"
+
+        archive_size = int(response.headers["Content-Length"])
+        assert archive_size > 0
+
+        chunk_size = 2 ** 10
+        with tqdm(
+            unit="B",
+            unit_scale=True,
+            unit_divisor=chunk_size,
+            desc="Downloading ROMs",
+            leave=False,
+            total=archive_size,
+        ) as pbar:
+            buffer = io.BytesIO()
+            for chunk in response.iter_content(chunk_size=chunk_size):
```

Comment:
Thanks for figuring this out. This greatly improves download speed, and is a huge improvement.